### PR TITLE
ENH: Setup prep

### PR DESCRIPTION
**Pull request by jklenzing**
Added basic features to allow pysatModelUtils to function independently with pysat as a requirement.  Features include:
- setup file
- travis config (not currently functional)
- Documentation for contributing to the code

@rstoneback, I don't think I have permission to set up Travis CI for this account.

### CODE REVIEW

**Comment by jklenzing** [comment]
unit tests are passing locally using `pytest -vs`.

**Comment by jklenzing** [comment]
Install using `python setup.py install`

**Comment by rstoneback** [comment]
Thanks Jeff. I’ll try to get to this today after class and in between meetings.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+language: python
+dist: xenial
+matrix:
+  include:
+    - python: 2.7
+    - python: 3.5
+    - python: 3.6
+    - python: 3.7
```
Comment (jklenzing) [inline .travis.yml]: Based on my sami2py streamlined setup.  We can pull the other tests out once we get Travis CI set up if needed.  Tests take about 2 min run locally.

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+before_install:
+  - pip install pytest-cov
+  - pip install coveralls
+  - pip install future
+
+install:
+  - python setup.py install
+
+script:
+  - pytest -vs --cov=pysatModelUtils/
```
Comment (jklenzing) [inline .travis.yml]: Experimenting with pytest here, but could run nose.  Tests are written the same, just a more streamlined report in the end.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Change Log
+All notable changes to this project will be documented in this file.
+This project adheres to [Semantic Versioning](http://semver.org/).
+
+## [0.1.0] - 2019-xx-xx
```
Comment (jklenzing) [inline CHANGELOG.md]: Planning to start the spinoff modules as 0.1.0 for basic versions that perform the same functions as the pysat versions, then increment as they become more mature.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    version = version_file.read().strip()
+
+# change setup.py for readthedocs - commented for now
+# on_rtd = os.environ.get('READTHEDOCS') == 'True'
+if sys.version_info.major == 2:
+    install_requires = ['xarray<0.12', 'pandas>=0.23, <0.25',
+                        'numpy>=1.12, <1.17', 'scipy<1.3', 'pysat']
+else:
+    install_requires = ['xarray', 'pandas>=0.23, <0.25', 'numpy>=1.12',
+                        'pysat']
```
Comment (jklenzing) [inline setup.py]: system requirements copied from pysat for now.  Will refine before 0.1.0 release

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+
+# Run setup
+
+setup(name='pysatModelUtils',
+      version=version,
+      url='github.com/pysat/pysatModelUtils',
+      author='Angeline G. Burrell, Russell Stoneback',
+      author_email='angeline.burrell@nrl.navy.mil',
+      description='Seasonal analysis routines for pysat',
```
[the remark this answers is not in the document]
Comment (jklenzing) [reply setup.py]: Good catch.

**Comment by jklenzing** [comment]
Travis is back!  It is currently not running the code because the pysat dependency fails when installing pysatMagVect.

File: CHANGELOG.md (hunk 3 above)
Comment (rstoneback) [reply CHANGELOG.md]: Fair

File: .travis.yml (hunk 2 above)
Comment (rstoneback) [reply .travis.yml]: Shows as outdated, but we shouldn't stick with nose if it isn't active.

Comment (jklenzing) [reply .travis.yml]: I'm open to either.  Testing nose now to see where the issues are.

**Comment by jklenzing** [comment]
Running into all errors within Travis CI environment.  Tests run fine locally.  Keep getting this on Travis:
```
ModuleNotFoundError: No module named 'pysat.instruments.pysat_testing'
```

**Comment by rstoneback** [comment]
We still have a few more months on 2.7...
Incidentally, it is still listed as supported in setup.py

**Comment by jklenzing** [comment]
Yeah, this is in flux.  Been turning things on and off to try to fix.

**Comment by rstoneback** [comment]
OK, no worries then.

I also don't see pysat.instruments.pysat_testing

on my local install of branch calc_freq_fix

**Comment by rstoneback** [comment]
Friendly fyi that pysat_testing isn't listed in instruments.init, so I wouldn't see it by default in the loaded methods.

**Comment by jklenzing** [comment]
Current working theory is that this is related to the pysat install.  Using `pip install python` simply grabs an old version that is not compatible with travis.  Doing a manual install from the develop branch leads to a billion dependency issues.  Trying again with a simplified version that removes sgp4 and its extra dependencies.

**Comment by rstoneback** [comment]
The pysat Travis file used conda to create the python environment. Or at least it used to. You’ve been busy. May be worth using again. I remember I had to grab a template Travis file from the internet to get things going initially with pysat.

**Comment by jklenzing** [comment]
Back to green!

**Comment by jklenzing** [comment]
OK, so we've hit a catch-22.  We need the core pysat in order to run unit tests on codes that have pysat as a dependency.  This is because the dependencies are complex enough that something is likely to go wrong, resulting in the test files being unable to load a test instrument.

Since we can't (or at least shouldn't) remove these functions from pysat yet, I've created a separate branch where pysat_sgp4 and its dependencies have been removed.  This takes pyglow, apexpy, etc, out of the equation.  It's a temporary solution, but it will help get everything sorted out.

**Comment by jklenzing** [comment]
OK, this one is done.  `pytest` and `coverage` are both working, as are Travis CI and Coveralls.  Let me know if you have comments!

File: .travis.yml (hunk 2 above)
Comment (jklenzing) [reply .travis.yml]: Fixed!

File: description.txt [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+Main Features
+-------------
+- Comparison of models and measured data.
+- Matching of instruments to modelled data parameters.
+- Extraction of instrument-aligned data from a modelled data set.
+
+Documentation
+---------------------
+`Full Documentation <http://pysat.readthedocs.io/en/latest/>`_
```
Comment (rstoneback) [inline description.txt]: Link to pysat documentation - we should probably note it is for pysat only

File: pysatModelUtils/tests/test_model_utils.py [hunk 7]
```diff
@@ -1,18 +1,13 @@
...
-from pysat import model_utils as mu
+import pysatModelUtils as mu
 
 
 class TestBasics():
     def setup(self):
         """Runs before every method to create a clean testing setup."""
-        self.testInst = pysat.Instrument(platform='pysat',
-                                         name='testing',
+        self.testInst = pysat.Instrument('pysat', 'testing',
```
Comment (rstoneback) [inline pysatModelUtils/tests/test_model_utils.py]: as a matter of curiosity, why remove the platform, name keywords?

Comment (jklenzing) [reply pysatModelUtils/tests/test_model_utils.py]: This was back in my "try anything to fix" stage. Will fix.

File: description.txt (hunk 6 above)
Comment (jklenzing) [reply description.txt]: fixed

**Pull request merged**